account-service/src/main/java/com/bdonor/accountservice/Controller/AccountController.java

```diff
@@ -1,76 +1,111 @@
 package com.bdonor.accountservice.Controller;
 
-import com.bdonor.accountservice.Models.AccountHelper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
 import com.bdonor.accountservice.Models.User;
-import com.bdonor.accountservice.Repository.UserRepository;
+import com.bdonor.accountservice.Repository.DynamoRepo;
 import com.google.gson.Gson;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
-import org.springframework.stereotype.Controller;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.web.bind.annotation.*;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
 
-@Controller
-@Component
+
+@RestController
 public class AccountController {
 
     @Autowired
-    private AccountHelper Service_functions;
-
-    @ResponseBody // Works
-    @GetMapping(value = "/create/{bloodGroup}/{firstname}/{surname}/{email}/{password}/{addressline}/{postcode}")
-    public String create( @PathVariable String bloodGroup , @PathVariable  String firstname, @PathVariable  String surname, @PathVariable  String email, @PathVariable  String password, @PathVariable  String addressline, @PathVariable  String postcode){
-        User CreateUser = Service_functions.create(bloodGroup, firstname,  surname,  email,  password,  addressline,  postcode);
-        System.out.println("this is working");
-        return CreateUser.toString();
-    }
+    private DynamoRepo dynamoRepo;
 
-    @ResponseBody
-    @GetMapping("/getUser/{firstname}") // Works Partially - Only works for one user within database, if there are more with the same name, error is given
-    public String getUser( @PathVariable String firstname ){
-        System.out.println("Working");
-        return Service_functions.getByfirstName(firstname).toString();
-    }
+    @Autowired
+    private DynamoDBMapper mapper;
+
+    @Autowired
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
 
     @ResponseBody
-    @GetMapping("/getAll")
-    public String getAllUsers(){ // Works
-        System.out.println("This Works");
+    @PostMapping(value = "/create/{bloodGroup}/{firstName}/{lastName}/{email}/{password}/{addressline}/{postcode}")
+    public String Register( @PathVariable String bloodGroup , @PathVariable  String firstName, @PathVariable  String lastName, @PathVariable  String email, @PathVariable  String password, @PathVariable  String addressline, @PathVariable  String postcode){
+        User checkEmail = mapper.load(User.class, firstName, email);
```

Comment:
https://github.com/we-3hackathon/bdonor/blob/e81f5b4d18581c56ab3e031675033d7a09cb62ef/account-service/src/main/java/com/bdonor/accountservice/Controller/AccountController.java#L36-L76

Part of business logic, to be abstracted under **Model** package